$NetBSD$

Find the separate iconv library. Store an absolute runtime locale path.
The iconv change adapts pkgsrc patch-ac for Meson.
Original: patch-ac,v 1.3 2017/09/02 22:45:04 leot Exp.
Prepared with AI assistance for EmberBSD.

--- meson.build.orig
+++ meson.build
@@ -22,6 +22,9 @@
   libintl_dep = cc.find_library('intl', required: true)
 endif
 
+# iconv can require a separate library on BSD systems.
+iconv_dep = dependency('iconv')
+
 systemd_dep = dependency('systemd', required: false)
 
 #
@@ -46,7 +49,7 @@
 }
 '''
 
-if cc.compiles(iconv_const_check, name: 'iconv with const')
+if cc.compiles(iconv_const_check, dependencies: iconv_dep, name: 'iconv with const')
   conf_data.set('ICONV_CONST', 'const')
 else
   conf_data.set('ICONV_CONST', '')
@@ -54,7 +57,7 @@
 
 # Set install directories
 conf_data.set_quoted('XDGCONFDIR', get_option('sysconfdir') / 'xdg')
-conf_data.set_quoted('LOCALEDIR', get_option('localedir'))
+conf_data.set_quoted('LOCALEDIR', get_option('prefix') / get_option('localedir'))
 
 # Generate config.h
 configure_file(
